WordPress/Sniffs/Theme/NoFaviconSniff.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Check for hardcoded favicons instead of using core implementation.
+ *
+ * @link    https://make.wordpress.org/themes/handbook/review/required/#core-functionality-and-features
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_NoFaviconSniff implements PHP_CodeSniffer_Sniff {
+
+	const REGEX_TEMPLATE = '` (?:%s)`i';
+	const REGEX_ATTR_TEMPLATE = '%1$s=[\'"](?:%2$s)[\'"]';
+
+	protected $attribute_blacklist = array(
+		'rel' => array(
+			'shortcut',
+			'icon',
+			'apple-touch-icon',
+			'apple-touch-icon-precomposed',
+		),
```

Comment:
We should add:
* 'shortcut icon'
* 'bookmark icon'

Not sure about `'icon shortcut'` as I'm not sure that's even a valid value. If it is, it should be added to the list as well.

These could also be considered for addition (both in unit tests as well as in the list:
```
<meta name="apple-mobile-web-app-title" content="<?php echo self::APP_NAME; ?>" />
<meta name="msapplication-config" content="/my/path/browserconfig.xml" />
<meta name="msapplication-TileColor" content="#d6001e" />
<meta name="msapplication-TileImage" content="', esc_url( plugins_url( 'images/icons/mstile-144x144.png?v=' . self::VERSION, __FILE__ ) ), '" />
<meta name="msapplication-square70x70logo" content="', esc_url( plugins_url( 'images/icons/tiny.png?v=' . self::VERSION, __FILE__ ) ), '" />
<meta name="msapplication-square150x150logo" content="', esc_url( plugins_url( 'images/icons/square.png?v=' . self::VERSION, __FILE__ ) ), '" />
<meta name="msapplication-wide310x150logo" content="', esc_url( plugins_url( 'images/icons/wide.png?v=' . self::VERSION, __FILE__ ) ), '" />
<meta name="msapplication-square310x310logo" content="', esc_url( plugins_url( 'images/icons/large.png?v=' . self::VERSION, __FILE__ ) ), '" />
```